homeassistant/components/light/mochad.py

```diff
@@ -0,0 +1,94 @@
+"""
+Contains functionality to use a X10 dimmer over Mochad.
+
+For more details about this platform, please refer to the documentation at
+https://home.assistant.io/components/light.mochad
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, Light)
+from homeassistant.components import mochad
+from homeassistant.const import (CONF_NAME, CONF_PLATFORM)
+from homeassistant.helpers import config_validation as cv
+
+DEPENDENCIES = ['mochad']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ADDRESS = 'address'
+CONF_DEVICES = 'devices'
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): mochad.DOMAIN,
+    CONF_DEVICES: [{
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Required(CONF_ADDRESS): cv.x10_address,
+        vol.Optional('comm_type'): cv.string,
+    }]
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up X10 dimmers over a mochad controller."""
+    devs = config.get(CONF_DEVICES)
+    add_devices([MochadLight(
+        hass, mochad.CONTROLLER.ctrl, dev) for dev in devs])
+    return True
+
+
+class MochadLight(Light):
+    """Representation of a X10 dimmer over Mochad."""
+
+    def __init__(self, hass, ctrl, dev):
+        """Initialize a Mochad Light Device."""
+        from pymochad import device
+
+        self._controller = ctrl
+        self._address = dev[CONF_ADDRESS]
+        self._name = dev.get(CONF_NAME, 'x10_light_dev_%s' % self._address)
+        self._comm_type = dev.get('comm_type', 'pl')
+        self.device = device.Device(ctrl, self._address,
+                                    comm_type=self._comm_type)
+        self._brightness = 0
+        self._state = self._get_device_status()
```

Comment:
This is just being used to set the initial state, we actually don't want to rely on this method for state updates after the devices is initialized. Mochad doesn't actually know the devices state it's just tracking in memory the last command it received. Just like when I implemented the mochad switch in https://github.com/home-assistant/home-assistant/pull/3970 relying on mochad is for device state just makes things go out of sync and cause weird behavior. It was much more reliable to have home-assistant track the device state.